internal/toolexec/proxy/proxy.go

```diff
@@ -0,0 +1,120 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+type (
+	// CommandType represents a Go compilation command type
+	// Currently support types are:
+	// - compile
+	// - link
+	// - unknown
+	CommandType string
+	// Command represents a Go compilation command
+	Command interface {
+		Args() []string
+		ReplaceParam(param string, val string) error
+		Stage() string
+		Type() CommandType
+	}
```

Comment:
Can the interface be named/extracted? (It does not necessarily need to be exported)